src/spec-lang/tc/typecheck.ts

```diff
@@ -1539,30 +1493,29 @@ export function tcIndexAccess(expr: SIndexAccess, ctx: STypingCtx, typeEnv: Type
     throw new SWrongType(`Cannot index into the type ${baseT.pp()}`, expr, baseT);
 }
 
-/**
- * @todo Seems to need rework due to recent changes in solc-typed-ast.
- */
-export function tcMemberAccess(expr: SMemberAccess, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode {
-    const baseT = tc(expr.base, ctx, typeEnv);
+function getFieldOfBuiltinStruct(
+    struct: BuiltinStructType,
+    expr: SMemberAccess,
+    typeEnv: TypeEnv
+): TypeNode {
+    const type = struct.getFieldForVersion(expr.member, typeEnv.compilerVersion);
 
-    if (baseT instanceof BuiltinStructType) {
-        const typing = baseT.members.get(expr.member);
+    if (type === undefined) {
+        throw new SInaccessibleForVersion(expr, expr.member, typeEnv.compilerVersion);
+    }
 
-        if (typing === undefined) {
-            throw new SNoField(
-                `Builtin struct "${expr.base.pp()}" does not have a member "${expr.member}"`,
-                expr,
-                expr.member
-            );
-        }
+    return type;
+}
 
-        const type = getTypeForCompilerVersion(typing, typeEnv.compilerVersion);
+export function tcMemberAccess(expr: SMemberAccess, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode {
+    const baseT = tc(expr.base, ctx, typeEnv);
 
-        if (type === undefined) {
-            throw new SInaccessibleForVersion(expr, expr.member, typeEnv.compilerVersion);
-        }
+    if (baseT instanceof BuiltinStructType) {
+        return getFieldOfBuiltinStruct(baseT, expr, typeEnv);
+    }
 
-        return type;
+    if (baseT instanceof BuiltinFunctionType && baseT.returns[0] instanceof BuiltinStructType) {
```

Comment:
I am confused by this logic. `baseT` is a function type. So the base refers to the function itself, not to a function call. So even if `baseT` returns some builtin struct, we shouldn't be able to resolve fields on `baseT` right? Only if `baseT` was called.

Reply:
I guess this is edge-case for `type()` or so, that returns a builtin struct. So this is more like an edge case. Maybe there is better way of addressing it.

Reply:
The fix was to change how we type the `type` identifier in [021aa55](https://github.com/ConsenSys/scribble/pull/203/commits/021aa552739d19bc244aa72209ac5225f40f4d2c)